test/e2e-go/restAPI/simulate/simulateRestAPI_test.go

```diff
@@ -2694,6 +2694,52 @@ int 1
 		},
 	}
 	a.Equal(expectedResult, resp)
+
+	// PopulateResourceArrays=true should also work
+	resp, err = testClient.SimulateTransactions(v2.PreEncodedSimulateRequest{
+		TxnGroups: []v2.PreEncodedSimulateRequestTransactionGroup{
+			{
+				Txns: []transactions.SignedTxn{stxn},
+			},
+		},
+		AllowUnnamedResources:  true,
+		PopulateResourceArrays: true,
+	})
+	a.NoError(err)
+
+	otherAddressAddress := basics.Address{}
+	err = otherAddressAddress.UnmarshalText([]byte(otherAddress))
+	a.NoError(err)
+
+	expectedResult = v2.PreEncodedSimulateResponse{
+		Version:   2,
+		LastRound: resp.LastRound,
+		EvalOverrides: &model.SimulationEvalOverrides{
+			AllowUnnamedResources: &allowUnnamedResources,
+		},
+		TxnGroups: []v2.PreEncodedSimulateTxnGroupResult{
+			{
+				Txns: []v2.PreEncodedSimulateTxnResult{
+					{
+						Txn:               v2.PreEncodedTxInfo{Txn: stxn},
+						AppBudgetConsumed: &budgetUsed,
+					},
+				},
+				AppBudgetAdded:           &budgetAdded,
+				AppBudgetConsumed:        &budgetUsed,
+				UnnamedResourcesAccessed: &expectedUnnamedGroupResources,
+				PopulatedResourceArrays: map[int]simulation.PopulatedResourceArrays{
+					0: {
+						Accounts: []basics.Address{otherAddressAddress},
+						Assets:   []basics.AssetIndex{basics.AssetIndex(assetID)},
+						Apps:     []basics.AppIndex{basics.AppIndex(otherAppID)},
+						Boxes:    []logic.BoxRef{{App: basics.AppIndex(testAppID), Name: "A"}, {App: basics.AppIndex(0), Name: ""}},
+					},
+				},
+			},
+		},
+	}
+	a.Equal(expectedResult, resp)
```

Comment:
Re: the test failure that you're seeing. This is what the server sends back (decoded from the msgpack response):

```javascript
"unnamed-resources-accessed": {
  "app-locals": [
    {
      "account": "7WCFW7UO7NA3NTQWDPKMFT4NNTOU4A7M5IF5VGPB3OCH6TMLD4X6QX453M",
      "app": 1005
    }
  ],
  "asset-holdings": [
    {
      "account": "7WCFW7UO7NA3NTQWDPKMFT4NNTOU4A7M5IF5VGPB3OCH6TMLD4X6QX453M",
      "asset": 1002
    }
  ],
  "boxes": [
    {
      "app": 1006,
      "name:b64": "QQ=="
    }
  ],
  "extra-box-refs": 1
}
```

I don't think there's anything going wrong with the encoding or e2e aspect, but rather a logic error inside the simulate package when multiple different resource types are unspecified. I noticed that the test you're comparing against, `TestPopulateResources`, is much simpler and only tests accounts. I suggest replicating this in a unit test and debugging further there. Generally speaking the most complex test cases and corner cases should be unit tests in `simulation_eval_test.go`, then all the e2e test needs to do is make sure all inputs get picked up properly and all outputs are present as expected in the response (to catch encoding errors or fields being dropped).